### PR TITLE
Support to Regtest Address

### PR DESCRIPTION
**Pull request by crisdut**
Details: https://github.com/orgs/LNP-BP/discussions/128

### CODE REVIEW

**Comment by dr-orlovsky** [comment]
Is it that `AddressCompat` doesn't support regtest? If not, it should - since it is a part of the address string; and I will look how to update it first.

`AddressCompat` provides stronger guarantees that the address, once parsed, is supported by the wallet, that is why it is used here

**Comment by crisdut** [comment]
> Is it that `AddressCompat` doesn't support regtest? If not, it should - since it is a part of the address string; and I will look how to update it first.
> 
> `AddressCompat` provides stronger guarantees that the address, once parsed, is supported by the wallet, that is why it is used here

Hi @dr-orlovsky,

I reverted the change add created a new PR related:

- https://github.com/BP-WG/descriptor-wallet/pull/64

**Comment by dr-orlovsky** [comment]
Please see my comments in https://github.com/BP-WG/descriptor-wallet/pull/64#issuecomment-1326431925

We need to decide upon that problem first

File: src/lnpd/funding.rs [hunk 1]
```diff
@@ -296,11 +296,12 @@ impl FundingWallet {
     }
 
     pub fn next_funding_address(&self) -> Result<Address, Error> {
-        let address = DescriptorExt::<bitcoin::PublicKey>::address(
+        let spk = DescriptorExt::<bitcoin::PublicKey>::script_pubkey(
             &self.wallet_data.descriptor,
             &self.secp,
             &[UnhardenedIndex::zero(), self.wallet_data.last_normal_index],
         )?;
+        let address = Address::from_script(&spk, self.network).unwrap();
```
Comment (dr-orlovsky) [inline src/lnpd/funding.rs:304]: Sorry, again, `unwraps` are prohibited outside of the tests

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]

**Review by dr-orlovsky — APPROVED** [review]
ACK 226d2683e1b75dcd992b5ad520f503e284188dd9 as a temporary fix pending https://github.com/LNP-WG/lnp-node/issues/62

**Pull request merged**